src/components/CreateProject/ProjectForm/CreateProjectForm.tsx

```diff
@@ -28,7 +32,25 @@ const FormSchema = yup
   .required();
 
 export const CreateProjectForm = ({ dismiss }: Props) => {
-  const [createProject, { loading }] = useCreateProjectMutation();
+  const [createProject, { loading }] = useCreateProjectMutation({
+    update(cache, { data }) {
+      const newProject = data?.insert_project_one;
+      const prevProjects = cache.readQuery<ProjectListQuery>({
+        query: ProjectListDocument,
+      });
+
+      if (prevProjects && newProject) {
+        cache.writeQuery({
+          query: ProjectListDocument,
+          data: {
+            project_aggregate: {
+              nodes: [...prevProjects.project_aggregate.nodes, newProject],
+            },
+          },
+        });
+      }
+    },
```

Comment:
This is the data layer's concern, let's keep it for now but we should move it there later.